challenges/cloud/sre.md

```diff
@@ -0,0 +1,376 @@
+Kubernetes cluster. The choice of which Kubernetes cluster is up to you, but 
+please choose one that has the ability to run on macOS and Linux.
+
+## Key Dependencies
+
+You are welcome to solve this challenge using tools you are most familiar with,
+but, at a minimum, please include the following:
+
+* Go
+* make
+* Docker
+
+## Level 1
+
+### Server
+
+* HTTP API to retrieve the replica count of the Kubernetes Deployment
+
+### Automation
+
+* Ability to build the server in a Docker container
+* Ability to run the server
+
+## Level 2
+
+### Server
+
+* HTTP API to retrieve the replica count of the Kubernetes Deployment
+* HTTP API to set the replica count of the Kubernetes Deployment
+
+### Automation
+
+* Ability to build the server in a Docker container
+* Ability to run the server
+* Ability to execute integration tests against the local Kubernetes cluster
+
+## Level 3
+
+### Server
+
+* HTTP API to retrieve the replica count of the Kubernetes Deployment
+* HTTP API to set the replica count of a the Kubernetes Deployment
+* HTTP API to get the list of available Deployments in the Kubernetes cluster
+* HTTP health check verifying Kubernetes connectivity
+
+### Automation
+
+* Ability to build the server in a Docker container
+* Ability to run the server
+* Ability to execute integration tests against the local Kubernetes cluster
+* GitHub Actions for every commit that verifies code quality, ensures the software builds, and all tests pass
+
+### Deployment
+
+* Create a helm chart for the service
+
+## Level 4
+
+### Design
+
+In addition to including details about the requirements below, the design
+document should include a section describing how this solution could be deployed
+to an AWS EKS cluster.
+
+### Server
+
+* HTTP API to retrieve the replica count of the Kubernetes Deployment
+* HTTP API to set the replica count of a the Kubernetes Deployment
+* HTTP API to get the list of available Deployments in the Kubernetes cluster
+* HTTP health check verifying Kubernetes connectivity
+* Extend the API to support a difference between current and desired state
+* Extend the Server to support the ability to show the differences between current and desired state
+* Secure connections with mTLS
+
+### Automation
+
+* Ability to build the server in a Docker container
+* Ability to run the server
+* Ability to execute integration tests against the local Kubernetes cluster
+* GitHub Actions for every commit that verifies code quality, ensures the software builds, and all tests pass
+* Produce production ready releases (binaries and docker image)
+
+### Deployment
+
+* Create a helm chart for the service
+* Deploy releases of the API Server to a Kubernetes cluster (and possible dependencies)
+* Upgrade releases of the API Server to a Kubernetes cluster (and possible dependencies)
+* Include a production Deployment of this service, including but not limited to: Deployment, ServiceAccount, Service
```

Comment:
This would likely be done via the helm chart mentioned above, right?

If so, I recommend listing the minimal criteria required for the helm chart at Level 3 (maybe it's just a deployment, and then add whatever else it is we care about here at level 4).

Reply:
Yeah great point, thank you. I've condensed so it's much more clearer.

I don't _believe_ there were extra requirements from L3 -> L4 with respects to Helm (@gozer, correct me if I'm wrong) so both requirements are mirrored.